src/rust/engine/src/nodes.rs

```diff
@@ -901,10 +904,16 @@ impl Snapshot {
           // And then create a Snapshot.
           stats
             .and_then(move |_| {
-              context
-                .core
-                .snapshots
-                .create(&context.core.vfs, path_stats)
+              let context = context.clone();
+              let pool = context.core.pool.clone();
+              pool
+                .spawn_fn(move || {
+                  fs::Snapshot::from_path_stats(
```

Comment:
`from_path_stats` returns a `Future`, so no need to spawn it onto a pool: can:
```
stats
  .and_then(move |_| {
    fs::Snapshot::from_path_stats(
      ..
    )
  })
  .map_err(...)
```

Reply:
Done